library/blob.c

```diff
@@ -491,6 +695,23 @@ static int eblob_check_disk_one(struct eblob_iterate_local *loc)
 		}
 	}
 
+	if ((ctl->flags & EBLOB_ITERATE_FLAGS_VERIFY_CHECKSUM) &&
+	    !(dc->flags & BLOB_DISK_CTL_REMOVE) &&
+	    !(dc->flags & BLOB_DISK_CTL_UNCOMMITTED)) {
+		struct eblob_write_control wc;
+		memset(&wc, 0, sizeof(wc));
+		eblob_rctl_to_wc(&rc, &wc);
+		eblob_dc_to_wc(dc, &wc);
+
+		err = eblob_verify_checksum(bc->back, &dc->key, &wc);
+		if (err) {
+			eblob_dump_wc(bc->back, &dc->key, &wc, "eblob_check_disk_one: checksum verification failed", err);
+			eblob_mark_entry_removed_purge(bc->back, &dc->key, &rc);
```

Comment:
Do we really want to remove key if checksum verification has failed?

I have a failed setup where there are keys without checksum and without no-checksum bit in header, verification always says that checksum is invalid for those keys, if I accidentally run iterator with checksum verification, it will remove all these keys without giving a chance to fix it or recover.